src/core/meshcop/dataset_local.cpp

```diff
@@ -161,16 +161,19 @@ otError DatasetLocal::Save(const otOperationalDataset &aDataset)
 otError DatasetLocal::Save(const Dataset &aDataset)
 {
     const Timestamp *timestamp;
-    otError          error;
+    otError          error = OT_ERROR_NONE;
 
     if (aDataset.GetSize() == 0)
     {
-        error = Get<Settings>().DeleteOperationalDataset(IsActive());
+        // do not propagate error back
+        Get<Settings>().DeleteOperationalDataset(IsActive());
+        mSaved = false;
         otLogInfoMeshCoP("%s dataset deleted", mType == Tlv::kActiveTimestamp ? "Active" : "Pending");
     }
     else
     {
-        error = Get<Settings>().SaveOperationalDataset(IsActive(), aDataset);
+        error  = Get<Settings>().SaveOperationalDataset(IsActive(), aDataset);
+        mSaved = true;
```

Comment:
Should we do `SuccessOrExit()` here on error? (we do this after the `if-else` with a `SuccessOrExit(error)` line 177/180).
The difference is if `mSaved` should be set or not (if/when saving data set fails).